src/core/grid_based_algorithms/lb_interface.cpp

```diff
@@ -481,8 +481,22 @@ const Utils::Vector3d lb_lbfluid_get_ext_force_density() {
 }
 
 void lb_lbfluid_set_tau(double tau) {
-  if (tau <= 0)
-    throw std::invalid_argument("tau has to be positive.");
+  if (tau <= 0.)
+    throw std::invalid_argument("LB tau has to be positive.");
+  extern double time_step;
```

Comment:
you can then use the free function here for the same validation to reduce code duplication